controller/api/public/client.go

```diff
@@ -178,10 +178,9 @@ func newClient(apiURL *url.URL, httpClientToUse *http.Client) (pb.ApiClient, err
 }
 
 func NewInternalClient(kubernetesApiHost string) (pb.ApiClient, error) {
-	apiURL := &url.URL{
-		Scheme: "http",
-		Host:   kubernetesApiHost,
-		Path:   "/",
+	apiURL, err := url.Parse(fmt.Sprintf("http://%s/", kubernetesApiHost))
+	if err != nil {
+		return nil, err
 	}
```

Comment:
This change results in us validating the value of the `--api-addr` flag immediately when it is provided, rather than waiting for the first API request to fail and return an error. This makes the output of the `version` and `check` commands cleaner when the flag is misspecified.